### PR TITLE
[RDY] Touch support for pinch zoom and move map

### PR DESCRIPTION
**Pull request by mugmuggy**
I have access to a convertable device that I have tested quickly at this stage addressing touch controls.

Both pinch zoom and move map which is working mostly okay with this code.

What I don't have is a device specifically like that mentioned in #593 which may have this behave entirely different or interfere generally with use so would be nice to get this tested elsewhere.

### CODE REVIEW

File: CorsixTH/Lua/game_ui.lua [hunk 1]
```diff
@@ -600,6 +601,38 @@ function GameUI:onMouseUp(code, x, y)
   return UI.onMouseUp(self, code, x, y)
 end
 
+function GameUI:onMultiGesture(numfingers, dTheta, dDist, x, y)
+  -- only deal with 2 finger events for now
+  if numfingers == 2 then
+    -- calculate magnitude of pinch
+    local mag = math.abs(dDist)
+    if mag > 0.002 then
```
Comment (TheCycoONE) [inline CorsixTH/Lua/game_ui.lua]: Make this a variable (constant) defined near the top of the class. Let's avoid magic numbers.

File: CorsixTH/Lua/game_ui.lua [hunk 2]
```diff
@@ -600,6 +601,38 @@ function GameUI:onMouseUp(code, x, y)
...
+      -- pinch action - constant needs to be tweaked
+      self.current_momentum.z = self.current_momentum.z + dDist * 100
```
Comment (TheCycoONE) [inline CorsixTH/Lua/game_ui.lua]: 100 is another magic number, so put it in a variable too.

File: CorsixTH/Lua/game_ui.lua [hunk 3]
```diff
@@ -649,6 +682,10 @@ function GameUI:onTick()
       self.current_momentum.z = self.current_momentum.z * self.momentum
       self.app.world:adjustZoom(self.current_momentum.z)
     end
+    if self.multigesturemove.x ~= 0.0 then
```
Comment (TheCycoONE) [inline CorsixTH/Lua/game_ui.lua]: condition isn't necessary, just set them to 0.

File: CorsixTH/Lua/game_ui.lua [hunk 4]
```diff
@@ -600,6 +601,38 @@ function GameUI:onMouseUp(code, x, y)
   return UI.onMouseUp(self, code, x, y)
 end
 
+function GameUI:onMultiGesture(numfingers, dTheta, dDist, x, y)
```
Comment (TheCycoONE) [inline CorsixTH/Lua/game_ui.lua]: Please add documentation to new functions.

**Comment by TheCycoONE** [comment]
I'd still like to see this one tested more before it lands. I'll see if I can get my hands on any devices. I think @wolfy1339 has one?

**Comment by wolfy1339** [comment]
I don't have my Surface anymore. I now have a laptop with a touch screen (Razer Blade Stealth)
________________________________

**Comment by TheCycoONE** [comment]
I can't seem to generate those events on y linux chromebook, both the touchscreen and the touchpad only make normal mouse events.

@wolfy1339 testing even with the razer blade would be helpful.

**Comment by wolfy1339** [comment]
👍 📦

**Pull request merged**